pkg/anonymization/anonymizer.go

```diff
@@ -0,0 +1,308 @@
+	ipNetworkRegex    *regexp.Regexp
+}
+
+type ConfigProvider interface {
+	Config() *config.Controller
+}
+
+// NewAnonymizer creates a new instance of anonymizer with a provided config observer and sensitive data
+func NewAnonymizer(clusterBaseDomain string, networks []string) (*Anonymizer, error) {
+	networks = append(networks, "127.0.0.1/8")
+
+	cidrs, err := k8snet.ParseCIDRs(networks)
+	if err != nil {
+		return nil, err
+	}
+
+	var networksInformation []subnetInformation
+	for _, network := range cidrs {
+		lastIP := network.IP
+		networksInformation = append(networksInformation, subnetInformation{
+			network: *network,
+			lastIP:  lastIP,
+		})
+	}
+
+	return &Anonymizer{
+		clusterBaseDomain: strings.TrimSpace(clusterBaseDomain),
+		networks:          networksInformation,
+		translationTable:  make(map[string]string),
+		ipNetworkRegex:    regexp.MustCompile(Ipv4AddressOrNetworkRegex),
+	}, nil
+}
+
+// NewAnonymizer creates a new instance of anonymizer with a provided config observer and openshift config client
+func NewAnonymizerFromConfigClient(
+	ctx context.Context, kubeClient kubernetes.Interface, configClient configv1client.ConfigV1Interface,
+) (*Anonymizer, error) {
+	baseDomain, err := utils.GetClusterBaseDomain(ctx, configClient)
+	if err != nil {
+		return nil, err
+	}
+
+	networksConfig, err := configClient.Networks().Get(ctx, "cluster", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	var networks []string
+	for _, network := range networksConfig.Spec.ClusterNetwork {
+		networks = append(networks, network.CIDR)
+	}
+	for _, network := range networksConfig.Spec.ServiceNetwork {
+		networks = append(networks, network)
+	}
+	for _, network := range networksConfig.Spec.ExternalIP.AutoAssignCIDRs {
+		networks = append(networks, network)
+	}
+	for _, network := range networksConfig.Spec.ExternalIP.Policy.AllowedCIDRs {
+		networks = append(networks, network)
+	}
+	for _, network := range networksConfig.Spec.ExternalIP.Policy.RejectedCIDRs {
+		networks = append(networks, network)
+	}
+
+	clusterConfigV1, err := kubeClient.CoreV1().ConfigMaps("kube-system").Get(ctx, "cluster-config-v1", metav1.GetOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	if installConfig, exists := clusterConfigV1.Data["install-config"]; exists {
+		networkRegex := regexp.MustCompile(Ipv4NetworkRegex)
+		for _, network := range networkRegex.FindAllString(installConfig, -1) {
+			networks = append(networks, network)
+		}
+	}
+
+	// we're sorting by subnet lengths, if they are the same, we use subnet itself
+	utils.SortAndRemoveDuplicates(&networks, func(i, j int) bool {
+		if !strings.Contains(networks[i], "/") || !strings.Contains(networks[j], "/") {
+			return networks[i] > networks[j]
+		}
+
+		network1 := strings.Split(networks[i], "/")
+		network2 := strings.Split(networks[j], "/")
+
+		// subnet length
+		if network1[1] > network2[1] {
+			return true
+		} else if network1[1] < network2[1] {
+			return false
+		} else {
```

Comment:
This else statement is not necessary.

Reply:
Why? It is. If the length is equal, we use subnet

Reply:
If it doesn´t match if previous if and if-else it will always gonna be watched by the else... right?

So, you can simplify it by removing the else, and keeping the return. Something like:

```
// subnet length
if network1[1] > network2[1] {
    return true
} else if network1[1] < network2[1] {
    return false
}
return network1[0] > network2[0]
```

Reply:
If it doesn´t match if previous if and if-else it will always gonna be watched by the else... right?

So, you can simplify it by removing the else, and keeping the return. Something like:

```go
// subnet length
if network1[1] > network2[1] {
    return true
} else if network1[1] < network2[1] {
    return false
}
return network1[0] > network2[0]
```

Reply:
Hmm and what about:
```
if network1[1] != network2[1] {
   return network1[1] > network2[1]
}
return network1[0] > network2[0]
```

Reply:
@tremes even better. I think so.

Reply:
done